aws-crt/lib/aws-crt/io.rb

```diff
@@ -18,14 +18,25 @@ def initialize(max_threads = nil)
           # Ruby uses nil to request default values, native code uses 0
           max_threads = 0 if max_threads.nil?
 
-          @native = Aws::Crt.call do
+          native = Aws::Crt.call do
             Aws::Crt::Native.event_loop_group_new(max_threads)
           end
+
+          @native = FFI::AutoPointer.new(native, self.class.method(:on_release))
```

Comment:
Nice!  I do believe this addresses the issue finalize + release/close method and at least from what I read of the documentation, this usage (with `self.class.method`) should avoid issues of self reference that prevent GC.

Reply:
That `AutoPointer` class does some nice stuff to help with the finalizer, where it lets you pass a normal function and it creates the proc for you. It also lets you call `AutoPointer.free` to run the finalizer whenever you want.